File: valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        parentheses = []
        for i in s:
            if((i == "(")or(i=="[")or(i=="{")):
                parentheses.append(i)
            if (i == ")"):
                if(len(parentheses)==0):
                    return False
                if(parentheses.pop()!="("):
                    return False
            if (i=="]"):
                if(len(parentheses)==0):
                    return False
                if(parentheses.pop()!="["):
                    return False
            if (i=="}"):
                if(len(parentheses)==0):
                    return False
                if(parentheses.pop()!="{"):
                    return False
                    
        if(len(parentheses)!=0):
            return False
        return True

File: typescript/solutions/validParentheses.ts
export function isValid(s: string): boolean {
  if (s.length % 2 !== 0) return false;
  const stack: string[] = [];
  const openingBrackets = ["(", "{", "["];
  const sArray = s.split("");
  for (let i = 0; i < sArray.length; i++) {
    const element = sArray[i];
    const hasOpenBracket = openingBrackets.includes(element);
    if (hasOpenBracket) stack.push(element);
    else {
      if (stack.length === 0) return false;
      const openElement = stack.pop();
      if (openElement === "(" && element !== ")") return false;
      if (openElement === "{" && element !== "}") return false;
      if (openElement === "[" && element !== "]") return false;
    }
  }
  return stack.length === 0;
}
